### PR TITLE
[Merged by Bors] - chore: rename `Holder` to `Hoelder`

### PR DESCRIPTION
**Pull request by YaelDillies**
As indicated by https://en.wikipedia.org/wiki/Germanic_umlaut#Substitution, Hölder should be transcribed to Hoelder in ASCII.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by vihdzp — APPROVED** [review]
As a German learner, I approve of this.

File: Mathlib/Order/JordanHolder.lean [hunk 1]
```diff
@@ -407,7 +407,7 @@ theorem exists_last_eq_snoc_equivalent (s : CompositionSeries X) (x : X) (hm : I
         (second_iso_of_eq (isMaximal_eraseLast_last h0s)
             (sup_eq_of_isMaximal (isMaximal_eraseLast_last h0s) hm hetx) (by rw [inf_comm, htt]))
 
-/-- The **Jordan-Hölder** theorem, stated for any `JordanHolderLattice`.
+/-- The **Jordan-Hölder** theorem, stated for any `JordanHoelderLattice`.
 If two composition series start and finish at the same place, they are equivalent. -/
 theorem jordan_holder (s₁ s₂ : CompositionSeries X)
```
Comment (JasonKYi) [inline Mathlib/Order/JordanHolder.lean:412]:
```suggestion
theorem jordan_hoelder (s₁ s₂ : CompositionSeries X)
```

File: Mathlib/Topology/MetricSpace/Holder.lean [hunk 2]
```diff
@@ -41,199 +41,200 @@ open NNReal ENNReal Topology
...
 /-- A function `f : X → Y` between two `PseudoEMetricSpace`s is Hölder continuous with constant
 `C : ℝ≥0` and exponent `r : ℝ≥0` on a set `s : Set X`, if `edist (f x) (f y) ≤ C * edist x y ^ r`
 for all `x y ∈ s`. -/
-def HolderOnWith (C r : ℝ≥0) (f : X → Y) (s : Set X) : Prop :=
+def HoelderOnWith (C r : ℝ≥0) (f : X → Y) (s : Set X) : Prop :=
   ∀ x ∈ s, ∀ y ∈ s, edist (f x) (f y) ≤ (C : ℝ≥0∞) * edist x y ^ (r : ℝ)
 
 @[simp]
-theorem holderOnWith_empty (C r : ℝ≥0) (f : X → Y) : HolderOnWith C r f ∅ := fun _ hx => hx.elim
+theorem HoelderOnWith_empty (C r : ℝ≥0) (f : X → Y) : HoelderOnWith C r f ∅ := fun _ hx => hx.elim
```
Comment (JasonKYi) [inline Mathlib/Topology/MetricSpace/Holder.lean:59]:
```suggestion
theorem hoelderOnWith_empty (C r : ℝ≥0) (f : X → Y) : HoelderOnWith C r f ∅ := fun _ hx => hx.elim
```
Same with the theorems below

**Comment by dupuisf** [comment]
Shouldn't we also rename the file `Holder.lean`?

**Comment by YaelDillies** [comment]
Yes, but if I do that then the diff would get messed up. It will be a separate PR

**Comment by b-mehta** [comment]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**